src/app/handler/cart.js

```diff
@@ -0,0 +1,468 @@
+}
+
+function deleteCart (steamid) {
+    if (carts[steamid]) {
+        delete carts[steamid];
+    }
+}
+
+function getCart (steamid) {
+    if (!cartExists(steamid)) {
+        return undefined;
+    }
+
+    return carts[steamid];
+}
+
+exports.addToCart = function (steamID, sku, amount, deposit) {
+    const side = deposit ? 'their' : 'our';
+
+    const name = schemaManager.schema.getName(SKU.fromString(sku));
+
+    let message;
+
+    if (deposit) {
+        message = pluralize(name, amount, true) + ' ' + (amount > 1 ? 'have' : 'has') + ' been added to your cart';
+    } else {
+        // Get all items in inventory, we don't need to check stock limits for withdrawals
+        const amountCanTrade = inventory.getAmount(sku) - (cartExists(steamID) ? getCart(steamID).amount(sku, side) : 0);
+
+        // Correct trade if needed
+        if (amountCanTrade <= 0) {
+            message = 'I don\'t have any ' + pluralize(name, 0);
+            amount = 0;
+        } else if (amountCanTrade < amount) {
+            amount = amountCanTrade;
+            message = 'I only have ' + pluralize(name, amount, true) + '. ' + (amount > 1 ? 'They have' : 'It has') + ' been added to your cart';
+        } else {
+            message = pluralize(name, amount, true) + ' ' + (amount > 1 ? 'have' : 'has') + ' been added to your cart';
+        }
+    }
+
+    if (amount > 0) {
+        if (!cartExists(steamID)) {
+            createCart(steamID);
+        }
+
+        getCart(steamID).add(sku, amount, side);
+    }
+
+    return { cart: getCart(steamID).get(), message };
+};
+
+exports.removeFromCart = function (steamID, sku, amount, our, all = false) {
+    if (!cartExists(steamID)) {
+        return { cart: undefined, message: 'Your cart is empty' };
+    }
+
+    if (typeof sku === 'boolean') {
+        all = sku;
+        sku = undefined;
+    }
+
+    let message;
+
+    const name = sku ? schemaManager.schema.getName(SKU.fromString(sku)) : undefined;
+
+    const side = our ? 'our' : 'their';
+
+    if (all) {
+        message = 'Your cart has been emptied';
+        getCart(steamID).clear();
+    } else {
+        const whose = our ? 'my' : 'your';
+
+        const inCart = getCart(steamID).amount(sku, side);
+
+        if (inCart === 0) {
+            amount = inCart;
+            message = 'There are no ' + pluralize(name, amount) + ' on ' + whose + ' side of the cart';
+        } else if (amount > inCart) {
+            amount = inCart;
+            message = 'There were only ' + pluralize(name, amount, true) + ' on ' + whose + ' side of the cart. ' + (amount > 1 ? 'They have' : 'It has') + ' been removed';
+        } else {
+            message = pluralize(name, amount, true) + (amount > 1 ? ' have' : ' has') + ' been removed from ' + whose + ' side of the cart';
+        }
+    }
+
+    getCart(steamID).remove(sku, amount, side);
+
+    if (getCart(steamID).isEmpty()) {
+        deleteCart(steamID);
+        return { cart: undefined, message };
```

Comment:
`{ cart: undefined, message };` is this correct?

Reply:
It is correct yes

Reply:
Didn't know that was possible, cool